inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,348 @@
+                               keypath, value_to_check, position + 1)
+    return False
+
+
+# def get_actions(user_actions):
+#     class_actions = []
+#     conditions = []
+#     match_type_map = {
+#         'contains': 'contains',
+#         'is equal to': 'exact',
+#         'does not exist': 'missing',
+#         'matches regular expression': 'regex'
+#     }
+#     if not user_actions:
+#         return
+#
+#     for condition in user_actions.get('conditions', []):
+#         if not condition.get('key'):
+#             continue
+#         keypath = condition['key'].split('.')
+#         user_condition = {'value': condition['value'],
+#                           'keypath': keypath,
+#                           'match_type': match_type_map[condition['matchType']]}
+#         conditions.append(user_condition)
+#
+#     for user_action in user_actions.get('actions', []):
+#         if not user_action.get('mainKey'):
+#             continue
+#         keypath = user_action.get('mainKey').split('.')
+#         if not keypath:
+#             return
+#         if user_action.get('actionName') == 'Addition':
+#             class_actions.append(Addition(keypath=keypath, value=user_action.get('value'),
+#                                           match_type=match_type_map[user_action.get('matchType')],
+#                                           conditions=conditions))
+#         elif user_action.get('actionName') == 'Deletion':
+#             class_actions.append(Deletion(keypath=keypath, value=user_action.get('value'),
+#                                           value_to_check=user_action.get('updateValue'),
+#                                           match_type=match_type_map[user_action.get('matchType')],
+#                                           conditions=conditions))
+#         elif user_action.get('actionName') == 'Update':
+#             class_actions.append(Update(keypath=keypath, value=user_action.get('value'),
+#                                         match_type=match_type_map[user_action.get('matchType')],
+#                                         value_to_check=user_action.get('updateValue'),
+#                                         conditions=conditions))
+#     return class_actions
+#
+#
+# def process_records(user_actions, records, schema):
+#     class_actions = get_actions(user_actions)
+#     for record in records:
+#         for class_action in class_actions:
+#             class_action.apply(record, schema)
+#     return records
+#
+
+def compare_records(old_records, new_records, schema):
+    json_patches = []
+    errors = []
+    for index, new_record in enumerate(new_records):
+        json_patch = []
+        for dif in diff(old_records[index], new_record):
+            dif = list(dif)
+            path = force_list(dif[1])  # in case the path is a string and not an array
+            if dif[0] == 'add':
+                if not path[0]:
+                    path = force_list(dif[2][0][0])
```

Comment:
`dif[1][2][3][4]` looks terrible, I get that it's the library you're using but I suggest to wrap it and return a better format to use it here. 

so instead of `path = force_list(dif[2][0][0])` would be nice to see `path = diff_result.path`